projects/plugins/boost/app/assets/src/js/utils/cloud-css.ts

```diff
@@ -2,7 +2,43 @@
  * Internal dependencies
  */
 import api from '../api/api';
+import { getStatus, updateStatus, resetStatus, CloudCssStatus } from '../stores/cloud-css-status';
 
-export function requestCloudCss(): Promise< string > {
-	return api.post( '/cloud-css/request-generate' );
+export async function requestCloudCss(): Promise< void > {
+	// Todo: Debounce request.
+	resetStatus();
+	await api.post( '/cloud-css/request-generate' );
+	pollStatus( true );
+}
+
+export async function pollCloudCssStatus() {
```

Comment:
I'm not sure about this logic.

If I understand it correctly, `pollCloudCssStatus()` decides how fast it should poll Critical CSS status, and is called under two circumstances:
1. When you mount the Cloud Critical CSS Module panel.
2. When a pending Cloud Critical CSS task comes back as completed.

The interval timer is only ever cancelled in the second situation - when a pending request comes back as completed.

I can't see
1. How the interval timer gets sped up when you click "regenerate",
2. What prevents multiple timers from running side-by-side in the future if we add more pages to the dashboard, and users navigate to sub-pages - and return to the main page, re-mounting the Module components.

IMHO we should keep one interval timer id (maybe as a module-level variable) which we check and clear before setting up any new interval timers. I also feel that passing `true` or `false` to `pollStatus` is a little hard to follow given `pollStatus` then has logic in it to manually check for `status.pending` in it anyway.

In my opinion, these two methods could be merged - or, if we want to split them up - have one method which returns an appropriate interval to poll for given a `status` object?

Reply:
+1

Reply:
I agree with the tidy-up part, I will update and revert.

Meanwhile, a few info:
> `pollCloudCssStatus` is called under
> 2. When a pending Cloud Critical CSS task comes back as completed.

Not exactly. It is called when a generation request is completed so that it starts polling for updates given by the wpcom API.

> I can't see
> How the interval timer gets sped up when you click "regenerate",

Clicking `regenerate` calls `requestCloudCss` which resets the status and in turn changes the `pending` status to `true`. This makes the interval pick the shorter interval.

And, thanks for the review. Not clearing the interval in a loop is a major issue. I am fixing this.